From 58f45205bb11059d6997c490d8803e2020a7e9bf Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Wed, 7 Aug 2013 21:27:59 +0300
Subject: [PATCH] fix haddock markup

Haddock dislikes lines with '{- #///' and has special treatment for it.

> Haddock coverage:
>    0% (  0 /  1) in 'Bindings.Libgit2.Windows'
>    0% (  0 /  1) in 'Bindings.Libgit2.Version'
>    0% (  0 / 17) in 'Bindings.Libgit2.Common'
>    0% (  0 / 50) in 'Bindings.Libgit2.Errors'
>    0% (  0 /  3) in 'Bindings.Libgit2.Message'
>    0% (  0 /  8) in 'Bindings.Libgit2.Strarray'
>    0% (  0 /  5) in 'Bindings.Libgit2.Threads'
>
> Bindings/Libgit2/Types.hsc:39:1: parse error on input `{'

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
 Bindings/Libgit2/OdbBackend.hsc |  4 ++--
 Bindings/Libgit2/Transport.hsc  |  2 +-
 Bindings/Libgit2/Types.hsc      | 14 +++++++-------
 3 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/Bindings/Libgit2/OdbBackend.hsc b/Bindings/Libgit2/OdbBackend.hsc
index e212289..f7a8702 100644
--- a/Bindings/Libgit2/OdbBackend.hsc
+++ b/Bindings/Libgit2/OdbBackend.hsc
@@ -10,9 +10,9 @@ import Bindings.Libgit2.Types
 import Bindings.Libgit2.Oid
 import Bindings.Libgit2.Indexer
 {- struct git_odb_stream; -}
-{- #opaque_t git_odb_stream -}
+{-  #opaque_t git_odb_stream -}
 {- struct git_odb_writepack; -}
-{- #opaque_t git_odb_writepack -}
+{-  #opaque_t git_odb_writepack -}
 {- typedef int (* git_odb_foreach_cb)(const git_oid * id,
                                    void * payload); -}
 #callback git_odb_foreach_cb , Ptr (<git_oid>) -> Ptr () -> IO CInt
diff --git a/Bindings/Libgit2/Transport.hsc b/Bindings/Libgit2/Transport.hsc
index c32bb7e..44bb2d3 100644
--- a/Bindings/Libgit2/Transport.hsc
+++ b/Bindings/Libgit2/Transport.hsc
@@ -123,7 +123,7 @@ import Bindings.Libgit2.Types
 #num GIT_SERVICE_RECEIVEPACK_LS
 #num GIT_SERVICE_RECEIVEPACK
 {- struct git_smart_subtransport; -}
-{- #opaque_t git_smart_subtransport -}
+{-  #opaque_t git_smart_subtransport -}
 {- typedef struct git_smart_subtransport_stream {
             struct git_smart_subtransport * subtransport;
             int (* read)(struct git_smart_subtransport_stream * stream,
diff --git a/Bindings/Libgit2/Types.hsc b/Bindings/Libgit2/Types.hsc
index 2f6e7ee..5e8613a 100644
--- a/Bindings/Libgit2/Types.hsc
+++ b/Bindings/Libgit2/Types.hsc
@@ -36,17 +36,17 @@ import Bindings.Libgit2.Common
 {- typedef struct git_odb git_odb; -}
 #opaque_t git_odb
 {- typedef struct git_odb_backend git_odb_backend; -}
-{- #opaque_t git_odb_backend -}
+{-  #opaque_t git_odb_backend -}
 {- typedef struct git_odb_object git_odb_object; -}
 #opaque_t git_odb_object
 {- typedef struct git_odb_stream git_odb_stream; -}
-{- #opaque_t git_odb_stream -}
+{-  #opaque_t git_odb_stream -}
 {- typedef struct git_odb_writepack git_odb_writepack; -}
-{- #opaque_t git_odb_writepack -}
+{-  #opaque_t git_odb_writepack -}
 {- typedef struct git_refdb git_refdb; -}
 #opaque_t git_refdb
 {- typedef struct git_refdb_backend git_refdb_backend; -}
-{- #opaque_t git_refdb_backend -}
+{-  #opaque_t git_refdb_backend -}
 {- typedef struct git_repository git_repository; -}
 #opaque_t git_repository
 {- typedef struct git_object git_object; -}
@@ -70,7 +70,7 @@ import Bindings.Libgit2.Common
 {- typedef struct git_config git_config; -}
 #opaque_t git_config
 {- typedef struct git_config_backend git_config_backend; -}
-{- #opaque_t git_config_backend -}
+{-  #opaque_t git_config_backend -}
 {- typedef struct git_reflog_entry git_reflog_entry; -}
 #opaque_t git_reflog_entry
 {- typedef struct git_reflog git_reflog; -}
@@ -135,6 +135,6 @@ import Bindings.Libgit2.Common
 {- typedef struct git_push git_push; -}
 #opaque_t git_push
 {- typedef struct git_remote_head git_remote_head; -}
-{- #opaque_t git_remote_head -}
+{-  #opaque_t git_remote_head -}
 {- typedef struct git_remote_callbacks git_remote_callbacks; -}
-{- #opaque_t git_remote_callbacks -}
+{-  #opaque_t git_remote_callbacks -}
-- 
1.8.3.2

